src/main/js/portal/main/models/CartItem.ts

```diff
@@ -167,14 +168,15 @@ export default class CartItem {
 		return this._keyValPairs[key];
 	}
 
-	getNewUrl(keyVal: IdxSig){
+	getNewUrl(keyVal: IdxSig | PreviewSettings){
```

Comment:
I believe this is a more accurate type signature for this method, since it does not inherently have anything to do with `PreviewSettings`.
```suggestion
	getNewUrl(keyVal: Record<string, string | undefined>){
```

Reply:
Yes, since those could theoretically be any parameters to be passed to the `CartItem`'s URL, keeping it more generic is better. I have implemented this type change and made appropriate fixes for this.

(In practice, right now these are exclusively the same as `PreviewSettings`, with the additional `objId` parameter being used sometimes. I think another valid approach would be to go hard the other direction and eliminate the whole "`keyValPair` language and just use an extended `PreviewSettings` object, with `objId` added, instead. But doing so does explicitly couple the `Preview` with `CartItem` in a way that limits the ability for `CartItem`'s URL to be used for another purpose later on, which is perhaps not desirable. We can reapproach this later if we decide to revisit the structure of the portal project...)